src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -60,21 +65,6 @@ public CSharpType(Type type, bool isNullable, params CSharpType[] arguments)
             IsPublic = type.IsPublic && arguments.All(t => t.IsPublic);
         }
 
-        /// <summary>
-        /// Constructs a CSharpType that represents a union type.
-        /// </summary>
-        /// <param name="type">The type to convert.</param>
-        /// <param name="unionItemTypes">The list of union item types.</param>
-        /// <param name="isNullable">Flag used to determine if a type is nullable.</param>
-        public CSharpType(Type type, CSharpType[] unionItemTypes, bool isNullable) : this(
```

Comment:
This ctor is removed.
The reason why it is removed is:
1. this ctor is only for constructing a union type, where the `type` parameter here actually should only accept `BinaryData`, but does not really reflect on anywhere semantically.
2. this ctor is very alike to the other ctor constructing a type from a generic type with generic parameters, the only difference between them is one of them has `isNullable` on 2nd place the other has it on 3rd place, which is very confusing and easy to misuse.

This ctor is replaced by a static method `FromUnion` just like we have `FromLiteral`.